wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -676,18 +681,26 @@ public void lockWalletIfNeeded() {
     /**
      * Removes all the data and restarts the app showing onboarding screen.
      */
+    @SuppressWarnings("ResultOfMethodCallIgnored")
     public void wipe(Context context) {
         log.info("Removing all the data and restarting the app.");
 
         resetBlockchain();
         wallet.shutdownAutosaveAndWait();
         stopBlockchainService();
 
-        //noinspection ResultOfMethodCallIgnored
         walletFile.delete();
         cleanupFiles();
         config.clear(true);
-        PinRetryController.clearPrefs();
+        PinRetryController.getInstance().clearPinFailPrefs();
+
+        File walletBackupFile = getFileStreamPath(Constants.Files.WALLET_KEY_BACKUP_PROTOBUF);
+        if(walletBackupFile.exists())
+            walletBackupFile.delete();
+
+        File blockChainFile = new File(getDir("blockstore", Context.MODE_PRIVATE), Constants.Files.BLOCKCHAIN_FILENAME);
+        if (blockChainFile.exists())
+            blockChainFile.delete();
```

Comment:
This may be unnecessary since the wallet file is deleted.

Deleting the wallet file will trigger a blockchain reset.

Reply:
We don't need to worry about the DMN List, since that is handled by the BlockchainService

Reply:
Ok, got it. I was wondering if this is needed or not.
Removing.

Reply:
Turns out this is needed.  We can fix this as part of the new wipe PR Story.